### PR TITLE
Env dependant tag creation

### LINKED ISSUE #23: Creating tag for .net is not working
Hello,

For starters, great extension! :D

So for the issue:

Im not sure what languages are supported for tag creation here, but .Net has a bit different process to get the tar file.
From what I've read in sc the 'tag creation' only packs whatever is in the folder and sends it into appwrite.

For .net we first need to create 'publish package' and only then we can create a tar to send into appwrite from this package.

So now:

Would it be possible to either recognize .net solutions or pass the default path for tag method get the files from?
The path that the package is sitting is 'bin/Debug/net5.0/linux-x64'.
So the solution would be, at least for me, if the folder contains a '*.csproj' check for 'bin/Debug/net5.0/linux-x64' and tar this folder.

The process for .net is like this:

dotnet publish --runtime linux-x64 --framework net5.0 --no-self-contained
tar -C bin/Debug/net5.0/linux-x64 -zcvf code.tar.gz publish

### PR DESCRIPTION
**Pull request by Pastajello**
Hello :D

So I wanted to have .net tag creation that i implemented it myself :D

The code could be improved Im pretty sure. For now I've had the occasion to test it on Windows.
Test for Mac could be nice. - Ill do it on 17.08

Im pretty shitty with js/ts.

So to publish and generate the tar that I wanted and to replicate the steps I've used 'child process' instead of using some libraries/pacakges.
If its a good way to do this I do not know.
Also, to make things easier for me I've added 1 more step in tag creation -> runtime selection, with this it could be made possible to get more ways for creating tags for other runtimes. Automatic discovery of runtime could be faulty. 
And I was to lazy to do this.

If You have some ideas/code changes feel free :D

### CODE REVIEW

**Comment by Miksier** [comment]
Also working on Mac :)

File: src/commands/functions/createTag.ts [hunk 1]
```diff
@@ -64,10 +65,32 @@ export async function createTag(item?: TagsTreeItem | Uri): Promise<void> {
         if (command === undefined) {
             return;
         }
+
+        // there is a  special way to publishing a tag for .net5.0
+        const envPick = await window.showQuickPick(
+            [".net5.0", "other"],
+            { placeHolder: "Select runtime" }
+        );
```
Comment (alexweininger) [inline src/commands/functions/createTag.ts:73]: Is there a good way to detect automatically if they are using .NET? If so, we might be able to skip this prompt.

**Review by alexweininger — COMMENTED** [review]
I think these changes look good overall. Do you know if there are any other runtimes that will need custom behavior like .NET?

File: src/commands/functions/createTag.ts (hunk 1 above)
Comment (Miksier) [reply src/commands/functions/createTag.ts:73]: Yes we could. We could chek if there is a *.csproj in the folder. Then the prject is using .NET
Now project can be using Net3.1 or Net5.0, there we would need to read the *.csproj and find out what version it is from there :(

**Comment by Miksier** [comment]
> 
> 
> I think these changes look good overall. Do you know if there are any other runtimes that will need custom behavior like .NET?

I've checked it quickly from demos:
python:
PIP_TARGET=./.appwrite pip install -r ./requirements.txt --upgrade --ignore-installed
tar -zcvf code.tar.gz

deno:
tar

nodejs:
npm install 
tar

kotlin:
kotlinc HelloWorld.kt -include-runtime -d HelloWorld.jar

java:
javac ./HelloWorld.java

dart:
dart pub get
tar -zcvf code.tar.gz project

php:
tar -zcvf code.tar.gz .

But from documentation it seems that some steps are necesarry for package-manager runtimes:

python(already included):
PIP_TARGET=./.appwrite pip install -r ./requirements.txt --upgrade --ignore-installed

dart:
PUB_CACHE=./.appwrite pub get

deno:
DENO_DIR=./.appwrite deno cache main.ts

ruby:
docker run --rm -v $(pwd):/app -w /app --env GEM_HOME=./.appwrite appwrite/env-ruby-2.7:1.0.2 bundle install

SOOO, in short. Most runtimes have some custom stuff :D but someone would have to test it

**Pull request closed without merging**